trikot-viewmodels-declarative/compose/src/main/kotlin/com/mirego/trikot/viewmodels/declarative/compose/viewmodel/VMDImage.kt

```diff
@@ -150,16 +156,27 @@ fun RemoteImage(
             .build()
     )
 
-    when (coilPainter.state) {
-        is AsyncImagePainter.State.Success -> Image(
-            painter = coilPainter,
-            modifier = modifier,
-            alignment = alignment,
-            colorFilter = colorFilter,
-            contentScale = contentScale,
-            contentDescription = contentDescription
-        )
-        else -> placeholder(placeholderImage, coilPainter.state)
+    when (val state = coilPainter.state) {
+        is AsyncImagePainter.State.Success -> {
+            val drawable = state.result.drawable
+            if (drawable !is BitmapDrawable || drawable.bitmap.allocationByteCount <= MAX_BITMAP_SIZE) {
+                Image(
+                    painter = coilPainter,
+                    modifier = modifier,
+                    alignment = alignment,
+                    colorFilter = colorFilter,
+                    contentScale = contentScale,
+                    contentDescription = contentDescription
+                )
+            } else {
+                Log.e(TAG, "Unable to load bitmap: size too large (${drawable.bitmap.allocationByteCount})")
+
+                if (displayPlaceholderOnError) {
+                    placeholder(placeholderImage, state)
+                }
+            }
+        }
+        else -> placeholder(placeholderImage, state)
```

Comment:
This "else" is an error too. Maybe it should use the displayPlaceholderOnError flag too? Or maybe this flag is useless and the new case should always use the placeholder, just like for "network errors" ?

Reply:
True we can call directly placeholder() and if the user doesn't provide one it will fallback to an empty composable anyway, which is the same behaviour as this flag